### ISSUE TITLE
SAGA Reclassify Values Tool broken - 'Index out of range' error in log  :bug

### ISSUE THREAD

**Issue by MattCanada** [qgis/QGIS#32006]
For QGIS 3.8

I cannot successfully run the tool using any of the 3 methods. The reclassification doesn't work and I cannot fill the nodata values either. Here is the log output:

Algorithm 'Reclassify values' starting…
Input parameters:
{ 'INPUT' : 'D:/Matt/Projects/Seine/site3ligra/Base/tourond.tif', 'MAX' : 237, 'METHOD' : 1, 'MIN' : 236, 'NEW' : 1, 'NODATA' : 0, 'NODATAOPT ' : True, 'OLD' : 0, 'OTHEROPT ' : False, 'OTHERS' : 0, 'RESULT' : 'C:/Users/info/Desktop/2.sdat', 'RETAB' : [None], 'RNEW' : 238, 'ROPERATOR' : 0, 'SOPERATOR' : 1, 'TOPERATOR' : 2 }

Traceback (most recent call last):
File "C:/OSGEO4~1/apps/qgis/./python/plugins\processing\algs\saga\SagaAlgorithm.py", line 282, in processAlgorithm
s = '{}\t{}\t{}\n'.format(values[i], values[i + 1], values[i + 2])
IndexError: list index out of range

Execution failed after 0.05 seconds

![1](https://user-images.githubusercontent.com/55809344/65632663-d7832100-df9f-11e9-97e7-a699aed587d8.jpg)

**Comment by roya0045** [comment]
Can you try with another output format just to check? Maybe a tif.

**Comment by nyalldawson** [comment]
@roya0045 Saga only supports sdat outputs, so that's not the issue here.

@MattCanada 
I'd strongly suggest using the native QGIS reclassification algorithms in place of the saga ones. Have you tried those?

**Comment by nyalldawson** [comment]
Actually the issue here is that your reclassification table is empty. So there's 2 things:

1. The saga algorithm provider shouldn't raise an exception itself here, it should pass the empty table to saga and let it handle it

2. Use the QGIS algorithm instead. It's just much nicer, stabler, and faster all round.

**Comment by MattCanada** [comment]
@nyalldawson  Thanks for the update. I had have success with this tool in older versions. I'll double check as I very well may be doing something incorrectly!

In 3.8 I have been setting the 'mode' to only use a single value. However when I do this, the tool complains that I haven't set operators for the range and table options. If I choose operators for them and then run the tool, it throws the index out of range error.

I have also gotten this same error (in 3.8) if I set the 'mode' to table and put in one reclass entry for it to use. An example of a table entry would be to reclass the range between 237 and 238 to 239.

As a side note, I prefer this tool because of the bottom option to 'replace nodata' values. Sometimes I don't even need to reclass a raster, but I want to change the nodata values into 0's. I've found this tool to be the easiest way to do this, and I have gotten it to work in older Q versions.

This is a separate thing but it would be great to have another tool that only does this operation in isolation!
